Gemfile

```
@@ -34,11 +34,17 @@ group :development do
   gem "listen", ">= 3.0.5", "< 3.2"
   gem "spring"
   gem "spring-watcher-listen", "~> 2.0.0"
+  gem "rspec-rails"
+  gem "factory_bot_rails"
 end
 
 group :test do
   gem "capybara", ">= 2.15"
   gem "selenium-webdriver"
   gem "chromedriver-helper"
+  gem "faker"
+  gem "guard-rspec"
```

tương tự 3 gem ở dòng 46, 47, 48

---

```suggestion
  gem "guard-rspec"
```
cho phép tự động & khởi chạy thông số một cách thông minh khi các tệp được sửa đổi.
```gem "launchy" '``` là lớp trợ giúp để khởi chạy các ứng dụng đa nền tảng một cách dễ dàng 
```gem "shoulda-matchers"``` giúp cho việc viết test dễ dàng hơn, tiết kiệm được thời gian khi viết các test dài và phức tạp, code ngắn gọn, dễ đọc.

---

- gem để làm gì?
- cách hoạt động của gem?
- một số methods/helper method hỗ trợ?

---

gem "shoulda-matchers" : 
- giúp cho việc viết test dễ dàng hơn, tiết kiệm được thời gian khi viết các test dài và phức tạp, code ngắn gọn, dễ đọc.
- Cung cấp một số phương thức: 
+ActiveModel: have_secure_password, validate_absence_of,validate_confirmation_of...
+ ActiveRecord : accept_nested_attributes_for, belong_to, have_many, have_one....
+Actioncontroller.
gem "launchy"
- là lớp trợ giúp để khởi chạy các ứng dụng đa nền tảng một cách dễ dàng
- Khi Launchy được cài đặt, khi rspec được chạy lại, nó sẽ khởi chạy một phiên bản chưa được chỉnh sửa của trang cụ thể. Nó có thể đặc biệt hữu ích khi gỡ lỗi trong các bài kiểm tra tích hợp.